### ISSUE TITLE
Power operator should be performed right-to-left rather than left-to-right

### ISSUE THREAD

**Issue by Chaosus** [godotengine/godot#88902]
### Tested versions

a586e860e5fc382dec4ad9a0bec72f7c6684f020

### System information

Windows 11

### Issue description

The power operator seems to be executed in not the common order of that operation. In the Python the order of this operation follows right-to-left: 1 ** (2 ** 3) while in Godot currently it's executed left-to-right like (1 ** 2) ** 3. The fix should be performed both in GDScript and the expression parser.

### Steps to reproduce

Executing the following code line will emit error
```
assert(2 ** 2 ** 3 == 256)
```
In Python that code will be evaluated to true, while in Godot the result is 64, which is incorrect.

### Minimal reproduction project (MRP)

Too easy to reproduce

**Comment by bruvzg** [comment]
There's a note about it in the [documentation](https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_basics.html), so this might be intentional.

> The ** operator is [left-associative](https://en.wikipedia.org/wiki/Operator_associativity). This means that 2 ** 2 ** 3 is equal to (2 ** 2) ** 3. Use parentheses to explicitly specify precedence you need, for example 2 ** (2 ** 3).

**Comment by akien-mga** [comment]
From a quick check, this seems to be a situation where there isn't a standard.

https://en.wikipedia.org/wiki/Exponentiation#Identities_and_properties says:
![image](https://github.com/godotengine/godot/assets/4701338/a1ab6745-20a0-47b5-be97-bf4100a1fb1a)

My calculator (speedcrunch) handles it the same:
![image](https://github.com/godotengine/godot/assets/4701338/d25c4b9a-6633-490e-bcc0-c2cd12379dc5)

But then this page specifically points out there's no standard for serial exponentiation operators like `^`:
https://en.wikipedia.org/wiki/Order_of_operations#Serial_exponentiation

![image](https://github.com/godotengine/godot/assets/4701338/33dfad34-c551-480b-bba9-600d872b22b6)

So currently we're consistent with Excel and MATLAB, but not consistent with Python or Wolfram Alpha.

Given that changing it will break compat, we may need to just commit to the current implementation which is documented as left-associative. Or do a thorough review of what other programming languages do to see if there's a strong prevalence for right-associative.

**Comment by Chaosus** [comment]
Nvm, I was just looking through Discord and found that some users struggling with it:

![изображение](https://github.com/godotengine/godot/assets/3036176/2813faae-4441-4c17-a32a-101ac7da5e5f)

**Comment by ajreckof** [comment]
Didn't even know there was place where it was left associative as doing `( 2 ** 3) ** 4` is the same as doing ˋ2 ** (3 * 4)ˋ which imo is more readable but I agree that we shouldn't change it as a lot of user might be relying on this and would be too much of a compact breakage. I still think we should review wether to change it or not for Godot 5 when the time comes.

**Comment by dalexeev** [comment]
> There's a note about it in the [documentation](https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_basics.html), so this might be intentional.

I think this was not intentional, since this was not discussed in #58873. It was probably implemented this way because the parser does not currently provide associativity, all binary operators are left-associative. But now it's set in stone and should not be changed, at least until 5.0.

**Comment by Chaosus** [comment]
> It was probably implemented this way because the parser does not currently provide associativity, all binary operators are left-associative.

Yeah, as I far as I remember I did not figure out how to do it on GDScript part, and too now. On the expression parser side, it's relative easily to change.
